fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/StandardIndexMaintainer.java

```diff
@@ -363,10 +372,20 @@ public <M extends Message> List<IndexEntry> filteredIndexEntries(@Nullable final
         if (savedRecord == null) {
             return null;
         }
+        IndexMaintenanceFilter filter = state.filter;
+        if (filter == null) {
+            final IndexPredicate predicate = state.index.getPredicate();
+            if (predicate != null) {
+                return filteredIndexEntriesByPredicate(savedRecord);
+            } else {
+                filter = IndexMaintenanceFilter.NORMAL;
```

Comment:
here we set it back to `NORMAL`? I think it's doing something subtle but I think this is lost without a good comment somewhere.

Reply:
It is explained in the PR comment. I'll add a comment here too and refresh your review request .
✅